pkg/controller/storagecluster/noobaa_system_reconciler.go

```diff
@@ -42,63 +32,38 @@ func (r *ReconcileStorageCluster) ensureNoobaaSystem(sc *ocsv1.StorageCluster, r
 		}
 		return err
 	}
-
-	if foundCeph.Status.State == cephv1.ClusterStateCreated {
-		cephClusterCreated = true
+	if foundCeph.Status.State != cephv1.ClusterStateCreated {
+		reqLogger.Info("Waiting on ceph cluster to initialize before starting noobaa")
+		return nil
 	}
 
-	// check if this noobaa instance aleady exists
-	found := &nbv1.NooBaa{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nb.ObjectMeta.Name, Namespace: sc.Namespace}, found)
+	// Take ownership over the noobaa object
+	nb := &nbv1.NooBaa{}
+	err := controllerutil.SetControllerReference(sc, nb, r.scheme)
 	if err != nil {
-		if errors.IsNotFound(err) {
-
-			if cephClusterCreated {
-				// Arbitrary sleep to hopefully allow the
-				// CephCluster time to quiesce.
-				// TODO: Remove this
-				time.Sleep(5 * time.Second)
-				// noobaa system not found - create one
-				reqLogger.Info("Creating NooBaa system")
-				err := r.client.Create(context.TODO(), nb)
-				if err != nil {
-					reqLogger.Error(err, "Failed to create NooBaa system")
-					return err
-				}
-			} else {
-				reqLogger.Info("Waiting on ceph cluster to initialize before starting noobaa")
-				return nil
-			}
-		} else {
-			// other error. fail reconcile
-			reqLogger.Error(err, "Failed to get NooBaa system")
-			return err
-		}
-	} else {
-		// Update NooBaa CR if it is not in the desired state
-		if !reflect.DeepEqual(nb.Spec, found.Spec) {
-			reqLogger.Info("Updating spec for NooBaa")
-			found.Spec = nb.Spec
-			err := r.client.Update(context.TODO(), found)
-			if err != nil {
-				reqLogger.Error(err, "Failed to update NooBaa system")
-				return err
-			}
-		}
+		return nil
```

Comment:
Why not 
```suggestion
		return err
```
Adding a comment would be helpful is this is desired.

Reply:
fixed